<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="weights" default="weights/FastSAM-x.pt" />
  <arg name="image_size" default="1024" />
  <arg name="confidence_threshold" default="0.4" />
  <arg name="iou_threshold" default="0.9" />
  <arg name="device" default="cuda" />

  <node name="fastsam" pkg="fastsam_ros" type="segment_anything.py" output="screen">
    <param name="weights" value="$(arg weights)" />
    <param name="image_size" value="$(arg image_size)" />
    <param name="confidence_threshold" value="$(arg confidence_threshold)" />
    <param name="iou_threshold" value="$(arg iou_threshold)" />
    <param name="device" value="$(arg device)" />
  </node>
</launch>